### PR TITLE
[SPARK-37291][PYSPARK][FOLLOWUP] PySpark create SparkSession should pass initialSessionOptions

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
In this pr, when create SparkSession, we pass initialSessionOptions to SparkSession, to keep same code path with scala code.

### Why are the changes needed?
Keep same code path with scala code.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
existed UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @HyukjinKwon @dongjoon-hyun

File: sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -97,11 +97,11 @@ class SparkSession private(
    * since that would cause every new session to reinvoke Spark Session Extensions on the currently
    * running extensions.
    */
-  private[sql] def this(sc: SparkContext) = {
+  private[sql] def this(sc: SparkContext, initialSessionOptions: Map[String, String]) = {
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala:100]: Can we have another constructor: `private[sql] def this(sc: SparkContext) = this(sc, Map.empty)` instead?

File: python/pyspark/sql/session.py [hunk 2]
```diff
@@ -305,10 +305,7 @@ def __init__(
             ):
                 jsparkSession = self._jvm.SparkSession.getDefaultSession().get()
             else:
-                jsparkSession = self._jvm.SparkSession(self._jsc.sc())
-                if options is not None:
-                    for key, value in options.items():
-                        jsparkSession.sharedState().conf().set(key, value)
+                jsparkSession = self._jvm.SparkSession(self._jsc.sc(), self._jvm.PythonUtils.toScalaMap(options))
```
Comment (HyukjinKwon) [inline python/pyspark/sql/session.py:308]: Hm, is `self._jvm.PythonUtils.toScalaMap` required?

Comment (HyukjinKwon) [reply python/pyspark/sql/session.py:308]: I think you can just do:

```python
if options is not None:
     jsparkSession = self._jvm.SparkSession(self._jsc.sc(), options)
```

Comment (AngersZhuuuu) [reply python/pyspark/sql/session.py:308]: With `private[sql] def this(sc: SparkContext) = this(sc, Map.empty)` won't need such changes.

**Comment by HyukjinKwon** [comment]
Thanks for the followup!

File: python/pyspark/sql/session.py (hunk 2 above)
Comment (AngersZhuuuu) [reply python/pyspark/sql/session.py:308]: > Hm, is `self._jvm.PythonUtils.toScalaMap` required?

It's required
```
org.apache.spark.sql.SparkSession([class org.apache.spark.SparkContext, class java.util.HashMap]) does not exist
```

Comment (AngersZhuuuu) [reply python/pyspark/sql/session.py:308]: > I think you can just do:
>     if options is not None:
>           jsparkSession = self._jvm.SparkSession(self._jsc.sc(), options)

If so, we need to add a `if else`, why not just one line.

File: python/pyspark/sql/session.py [hunk 3]
```diff
@@ -305,10 +305,9 @@ def __init__(
             ):
                 jsparkSession = self._jvm.SparkSession.getDefaultSession().get()
             else:
-                jsparkSession = self._jvm.SparkSession(self._jsc.sc())
-                if options is not None:
-                    for key, value in options.items():
-                        jsparkSession.sharedState().conf().set(key, value)
+                jsparkSession = self._jvm.SparkSession(
+                    self._jsc.sc(), self._jvm.PythonUtils.toScalaMap(options)
```
Comment (HyukjinKwon) [inline python/pyspark/sql/session.py:309]: Looks good but last question. is this to ScalaMap required? We can make the Scala side to take Java map

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/session.py (hunk 3 above)
Comment (AngersZhuuuu) [reply python/pyspark/sql/session.py]: > Looks good but last question. is this to ScalaMap required? We can make the Scala side to take Java map

It seems worthwhile to use Java map to reduce one call of py4j.  Although Java map here looks not consistent.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master!

**Pull request closed without merging**